pkg/scheduler/scheduler_test.go

```diff
@@ -1070,6 +1070,177 @@ func TestEntryOrdering(t *testing.T) {
 	}
 }
 
+func TestLastSchedulingContext(t *testing.T) {
```

Comment:
Can you add more test scenarios?

With more than  two flavors.
When the admission is not possible (get to the end of the flavor list) 
What happens when the queue changes configuration (adds / removes flavors) between admission attempts.

Reply:
@trasc Hi, I have add some more test cases, including some test cases for scheduling context changed and for borrow before next flavor and borrow after all flavors. Please take a look, thank you!